libs/confman/testing/tests/file_storage_test.cpp

```diff
@@ -134,5 +158,118 @@ auto main(int argc, char** argv) -> int {
     ut::expect(called == 1);
   };
 
+  "backup on config change"_test = [&] {
+    std::filesystem::path const json_file_name{ file_name.parent_path() / "test_uuid.json" };
+
+    file_testable<test_me> conf{ ctx, json_file_name, test_me{ .a = observable<int>{ 3 }, .b = "bar" } };
+    conf.make_change()->a = 2;
+    ctx.run_for(std::chrono::milliseconds(1));
+    auto files = std::filesystem::directory_iterator(json_file_name.parent_path());
+
+    bool backup_found = false;
+    std::filesystem::path found_file;
+
+    for (const auto& file : files) {
+      if (file.path().string().find(static_cast<std::string>(json_file_name.parent_path().string() + "/test_uuid_")) !=
+          std::string::npos) {
+        backup_found = true;
+        found_file = file.path();
+        break;
+      }
+    }
+
+    ut::expect(backup_found) << "No backup file found for modified config";
+
+    if (backup_found) {
+      glz::json_t backup_json{};
+      std::string buffer{};
+      glz::read_file_json(backup_json, found_file.string(), buffer);
+
+      ut::expect((backup_json["a"].get<double>() - 3) < 0.01) << backup_json["a"].get<double>();
```

Comment:
`a` is an integer why not read as integer?

Reply:
Because `json_t` doesn't store `int`, it stores a variant `std::variant<null_t, double, std::string, bool, array_t, object_t>`

https://github.com/stephenberry/glaze/blob/main/include/glaze/json/json_t.hpp